src/event/inspect.rs

```diff
@@ -0,0 +1,74 @@
+use crate::event::{Event, LogEvent};
+use parking_lot::RwLock;
+use std::collections::HashMap;
+use tokio::sync::broadcast;
+
+type Result = ::std::result::Result<Event, ()>;
+type Sender = broadcast::Sender<LogEvent>;
+type Receiver = broadcast::Receiver<LogEvent>;
+
+pub struct EventInspector {
+    components: RwLock<HashMap<String, Sender>>,
+}
+
+impl EventInspector {
+    pub fn new() -> Self {
+        Self {
+            components: RwLock::new(HashMap::new()),
+        }
+    }
+
+    /// Associates a broadcast::Sender with a component name, so that subscribers may selectively
+    /// receive `LogEvent`s based on specific components. Returns a function that can be provided
+    /// to `.inspect` on an upstream LogEvent receiver channel.
+    pub fn adder(&self, component_name: String) -> impl FnMut(&Event) {
+        let mut lock = self.components.write();
+        let tx = lock.entry(component_name).or_insert_with(|| {
+            let (tx, _) = broadcast::channel(100);
```

Comment:
I chose a buffer size of `100` here to model the buffer from the upstream input rx. This should be sufficient, but we may want to play with the numbers as the choice was somewhat arbitrary.